packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -5,39 +5,193 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useRef, useContext, useState } from 'react';
-import { useSelect } from 'downshift';
+import React, { useRef, useContext, useState, FocusEvent, ForwardedRef, MouseEvent, ReactNode } from 'react';
+import { useSelect, UseSelectProps, UseSelectState } from 'downshift';
 import cx from 'classnames';
 import PropTypes from 'prop-types';
 import {
   Checkmark,
   WarningAltFilled,
   WarningFilled,
 } from '@carbon/icons-react';
-import ListBox, { PropTypes as ListBoxPropTypes } from '../ListBox';
+import ListBox, { ListBoxSize, ListBoxType, PropTypes as ListBoxPropTypes } from '../ListBox';
 import mergeRefs from '../../tools/mergeRefs';
 import deprecate from '../../prop-types/deprecate';
 import { useFeatureFlag } from '../FeatureFlags';
 import { usePrefix } from '../../internal/usePrefix';
 import { FormContext } from '../FluidForm';
+import { ReactAttr } from '../../types/common';
 
-const defaultItemToString = (item) => {
+const defaultItemToString = <ItemType,>(item?: ItemType): string => {
   if (typeof item === 'string') {
     return item;
   }
+  if (typeof item === 'number') {
+    return `${item}`
+  }
+  if (item !== null && typeof item === 'object'
+    && 'label' in item && typeof item['label'] === 'string') {
+    return item['label']
+  }
+  return '';
+};
+
+type ExcludedAttributes = 'id' | 'onChange';
+
+export interface OnChangeData<ItemType> {
+  selectedItem: ItemType | null;
+}
+
+export interface DropdownProps<ItemType>
+  extends Omit<ReactAttr<HTMLDivElement>, ExcludedAttributes> {
+
+  /**
+   * 'aria-label' of the ListBox component.
+   */
+  ariaLabel?: string;
+
+  /**
+   * Specify the direction of the dropdown. Can be either top or bottom.
+   */
+  direction?: 'top' | 'bottom';
+
+  /**
+   * Disable the control
+   */
+  disabled?: boolean;
+
+  /**
+   * Additional props passed to Downshift
+   */
+  downshiftProps: any; // UseSelectProps<ItemType>;
```

Comment:
is there a way to type this?

Reply:
@francinelucca thanks for spotting, added better type.